client/v2/autocli/common.go

```diff
@@ -67,22 +67,26 @@ func (b *Builder) buildMethodCommandCommon(descriptor protoreflect.MethodDescrip
 // options or the provided custom commands for each module. If the provided query command already contains a command
 // for a module, that command is not over-written by this method. This allows a graceful addition of autocli to
 // automatically fill in missing commands.
-func (b *Builder) enhanceCommandCommon(cmd *cobra.Command, moduleOptions map[string]*autocliv1.ModuleOptions, customCmds map[string]*cobra.Command, buildModuleCommand func(*cobra.Command, *autocliv1.ModuleOptions, string) error) error {
-	allModuleNames := map[string]bool{}
-	for moduleName := range moduleOptions {
-		allModuleNames[moduleName] = true
-	}
-	for moduleName := range customCmds {
-		allModuleNames[moduleName] = true
+func (b *Builder) enhanceCommandCommon(
+	cmd *cobra.Command,
+	appOptions AppOptions,
+	customCmds map[string]*cobra.Command,
+	buildModuleCommand enhanceCommandFunc,
+) error {
+	moduleOptions := appOptions.ModuleOptions
+	if len(moduleOptions) == 0 {
+		moduleOptions = map[string]*autocliv1.ModuleOptions{}
+		for name, module := range appOptions.Modules {
+			if module, ok := module.(HasAutoCLIConfig); ok {
+				moduleOptions[name] = module.AutoCLIOptions()
+			}
+		}
 	}
 
-	for moduleName := range allModuleNames {
+	for moduleName := range appOptions.Modules {
 		// if we have an existing command skip adding one here
-		if cmd.HasSubCommands() {
-			if _, _, err := cmd.Find([]string{moduleName}); err == nil {
```

Comment:
This was always returning true for simapp. We already had a helper that was working.